### PR TITLE
api: support failOpen in ext auth

### LINKED ISSUE #2897: Support failOpen in ext auth
ext authz setting is `failClosed` by default 
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto

Some users use ext auth as a way to enrich request headers, and prefer this external call to be `failOpen` (its okay to send the request to upstream/backend in case the API call fails/times out)

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by yeedove**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

- add a `failOpen` field for ext auth, set its default value to false

Fixes #2897

### CODE REVIEW

File: api/v1alpha1/ext_auth_types.go [hunk 1]
```diff
@@ -40,6 +40,15 @@ type ExtAuth struct {
...
 	HeadersToExtAuth []string `json:"headersToExtAuth,omitempty"`
+
+	// FailClosed is a switch used to control the behavior when a response from the External Authorization service cannot be obtained.
+	// If FailClosed is set to false, the system allows the traffic to pass through. Otherwise, if it is set to true or not set (defaulting to true),
+	// the system blocks the traffic and returns a HTTP 5xx error. This behavior signifies a secure-by-default approach,
+	// ensuring traffic is blocked when permissions cannot be confirmed by the External Authorization service.
+	//
+	// +optional
+	// +kubebuilder:default=true
+	FailClosed *bool `json:"failClosed,omitempty"`
```
[the remark this answers is not in the document]
Comment (zhaohuabing) [reply api/v1alpha1/ext_auth_types.go:51]: Yeah, `failOpen` feels better.

[the remark this answers is not in the document]
Comment (yeedove) [reply api/v1alpha1/ext_auth_types.go]: done

**Review by zhaohuabing — APPROVED** [review]
LGTM Thanks!

**Comment by piotrmsc** [comment]
quick question, here https://github.com/envoyproxy/gateway/blob/main/internal/xds/translator/extauth.go#L98 it is set explicitly to false, should this be changed as well?

**Pull request merged**